src/main/java/org/c02e/jpgpj/Encryptor.java

```diff
@@ -408,13 +411,68 @@ public void encrypt(InputStream plaintext, OutputStream ciphertext)
      * @throws PassphraseException if an incorrect passphrase was supplied
      * for one of the signing keys.
      */
-    public void encrypt(InputStream plaintext, OutputStream ciphertext,
-    FileMetadata meta) throws IOException, PGPException {
+    public void encrypt(
+        InputStream plaintext, OutputStream ciphertext, FileMetadata meta)
+            throws IOException, PGPException {
+        if (meta == null) {
+            meta = new FileMetadata();
+        }
+
+        try (OutputStream targetStream = prepareCiphertextOutputStream(ciphertext, meta)) {
+            // copy plaintext bytes into encryption pipeline
+            copy(plaintext, targetStream, null /* signer is inside the target */, meta);
+        }
+    }
+
+    public OutputStream prepareCiphertextOutputStream(File ciphertext)
+            throws IOException, PGPException {
+        // delete old output file
+        if (ciphertext.delete()) {
+            log.debug("prepareCiphertextOutputStream - deleted {}", ciphertext);
+        }
+
+        FileMetadata meta = new FileMetadata(ciphertext);
```

Comment:
the metadata is supposed to be for the original plaintext file (where applicable) -- for this method it could just always be null, or you could add another method parameter to allow the plaintext file's metadata to be passed in

Reply:
Will do...

Reply:
Done.